lib/modules/versioning/deb/index.ts

```diff
@@ -0,0 +1,154 @@
+import { regEx } from '../../../util/regex';
+import { GenericVersion, GenericVersioningApi } from '../generic';
+import type { VersioningApi } from '../types';
+
+export const id = 'deb';
+export const displayName = 'Deb version';
+export const urls = [
+  'https://www.debian.org/doc/debian-policy/ch-controlfields.html#version',
+  'https://manpages.debian.org/unstable/dpkg-dev/deb-version.7.en.html',
+];
+export const supportsRanges = false;
+
+const epochPattern = regEx(/^[0-9]+$/);
+const upstreamVersionPattern = regEx(/^[-+.:~A-Za-z0-9]+$/);
+const debianRevisionPattern = regEx(/^[+.~A-Za-z0-9]*$/);
+const numericPattern = regEx(/[0-9]+/g);
+const characterOrder =
+  '~ ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz+-.:';
+const numericChars = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9'];
+
+export interface DebVersion extends GenericVersion {
+  /**
+   * epoch, defaults to 0 if not present, are used to leave version mistakes and previous
+   * versioning schemes behind.
+   */
+  epoch: number;
+  /**
+   * upstreamVersion is the main version part: it defines the version of origin software
+   * that was packaged.
+   */
+  upstreamVersion: string;
+  /**
+   * debianRevsion is used to distinguish between different versions of packaging for the
+   * same upstream version.
+   */
+  debianRevision: string;
+}
+
+class DebVersioningApi extends GenericVersioningApi {
+  protected _parse(version: string): DebVersion | null {
+    /* Splitting version into "[epoch:]upstream-version[-debian-revision]"
+       All found numbers are exported as release info */
+    let epoch = 0;
+    let nonEpochVersion = version;
+    if (nonEpochVersion.includes(':')) {
+      const epochEnd = nonEpochVersion.indexOf(':');
+      const epochMatch = nonEpochVersion
+        .substring(0, epochEnd)
+        .match(epochPattern);
+      if (!epochMatch) {
+        return null;
+      }
+      epoch = parseInt(epochMatch[0], 10);
+      nonEpochVersion = nonEpochVersion.substring(epochEnd + 1);
+    }
+
+    let debianRevision = '';
+    let upstreamVersion;
+    if (nonEpochVersion.includes('-')) {
+      const revisionStart = nonEpochVersion.lastIndexOf('-');
+      debianRevision = nonEpochVersion.substring(revisionStart + 1);
+      upstreamVersion = nonEpochVersion.substring(0, revisionStart);
+    } else {
+      upstreamVersion = nonEpochVersion;
+    }
+
+    if (
+      !upstreamVersionPattern.test(upstreamVersion) ||
+      !debianRevisionPattern.test(debianRevision)
+    ) {
+      return null;
+    }
+    const release = [...nonEpochVersion.matchAll(numericPattern)].map((m) =>
+      parseInt(m[0], 10)
+    );
```

Comment:
```suggestion
    const release = nonEpochVersion.split('.').map(Number)
```

Reply:
The proposed change isn't doing the same thing: `dpkg` allows alphabetical characters in many places. The support this as best as possible in Renovate this extraction logic simply ignores any non-digit character (be it `.` or something else`:

- `v1.0.0` was parsed as `[1, 0, 0]`
- `2-0-0` as `[2, 0, 0]`
- `2.31-13+deb11u5` as `[2, 31, 13, 11, 5]`